client/go/outline/vpn/nm_polling.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpn
+
+import (
+	"fmt"
+	"time"
+)
+
+// TODO: NetworkManager is async, use signals instead of polling in the future
+const (
+	nmPollingCount = 20
+	nmPollingDelay = 50 * time.Millisecond
+)
+
+func nmPolling(doWork func() error) (err error) {
```

Comment:
I'd call this `retry`, since you are not polling a state. You are retrying the action until it succeeds.

Reply:
Renamed.